shopinvader_auth_jwt/components/service_context_provider.py

```diff
@@ -49,11 +49,11 @@ def _get_backend(self):
             backend_model = self.env["shopinvader.backend"]
             if backend:
                 # validate that this backend can be used for the aud
-                backend = backend if backend.jwt_aud == audience else backend_model
```

Comment:
According to https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.3, aud can be an array or a string, so we need to support both cases here ?

Reply:
do not know why, but keycloak puts something like this in its audience when I create new clients in v16.

Reply:
@hparfr yes, no problem with that, but I'm just saying the code must work wether the `aud` claim is a string or an array. So changing the test from `==` to  `in` will break compatibility.

Reply:
So we have: 
- `backend.jwt_aud ` string
-  `audience`  (array|string).

Since jwt_aud is a string, we can test if `audience` is `jwt_aud` (string == string) and if one of the values of `audience` is `jwt_aud` (string in array).

I think jwt_aud should stay a single value field (string). 


And yes my check is not good. If audience is a string then `in` return true for substrings. I need to change it to something   like : 

```py 
if isinstance(audience, list):
  backend = backend if audience in backend.jwt_aud else backend_model
elif isinstance(audience, str) 
  backend = backend if audience == backend.jwt_aud else backend_model
```

Do you agree ?

Reply:
Yes that is what I meant.

Actually, I'd do it earlier and make audiences plural:

```
audiences = self._jwt_payload.get("aud")
if isinstance(audiences, str):
   audiences = [audiences]
```